controllers/userController.js

```diff
@@ -1,13 +1,37 @@
-const { getUserById } = require('../services/userService')
+const { getUserById, getAllUsers, getUserByUsername} = require('../services/userService')
 
-async function getUser(ctx){
-    const username = ctx.params.username
-    const user = await getUserById(username)
 
+async function searchUser(username) {
 
-    const {password, ...data} = user
+    try {
+        const user = await getUserByUsername(username)
 
-    ctx.body = data
+        const {password, ...data} = user
+
+        return { data }
+
+    }catch (e) {
```

Comment:
you don't need try-catch block i container level at all;

Reply:
Done